### PR TITLE
[aggregator] tests for events and improve event v1 handling

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

- Added tests for event V1 and event V2 generation with aggregator snapshots. 
- Events V1 now also can use snapshots.

### CODE REVIEW

**Review by igor-aptos — APPROVED** [review]
fix looks good, so approving, but for testing - I think it would be better to use comparison way, with and without the flags, which makes sure transaction output doesn't change. that validates much more details than asserts can. Can you do that in a follow-up PR?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/events_with_aggregators.move [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    const E_COUNTER_ALREADY_EXISTS: u64 = 0;
+    const E_COUNTER_DOES_NOT_EXIST: u64 = 1;
+
+    struct EventV1 has drop, store {
+        value: AggregatorSnapshot<u64>,
+    }
+
+    #[event]
+    struct EventV2 has drop, store {
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/events_with_aggregators.move:15]: EventV2 has #[event] but not EventV1.
What's the difference? Is EventV1 still an event without the annotation?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+fn publish_test_package(h: &mut MoveHarness, aptos_framework_account: &Account) {
+    let path_buf = common::test_dir_path("aggregator_v2.data/pack");
+    assert_success!(h.publish_package_cache_building(aptos_framework_account, path_buf.as_path()));
+}
+
+fn create_test_txn(
+    h: &mut MoveHarness,
+    aptos_framework_account: &Account,
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:49]: Does this have to 0x1 account? Doesn't any account work?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+struct Counter {
+    value: Aggregator,
+}
+
+#[derive(Deserialize, Serialize)]
+struct Event {
+    value: AggregatorSnapshot,
+}
+
+fn publish_test_package(h: &mut MoveHarness, aptos_framework_account: &Account) {
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:42]: Do we have to publish the package in aptos framework? Doesn't any account work?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+fn check_events(outputs: Vec<TransactionOutput>) {
+    // We always emit fee statement event. Currently this is hardcoded
+    // for existing test case.
+    assert_eq!(outputs[0].events().len(), 1);
+    assert_eq!(outputs[1].events().len(), 1);
+
+    assert_eq!(outputs[2].events().len(), 2);
+    let event = assert_ok!(bcs::from_bytes::<Event>(
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:98]: Should this be `bcs::from_bytes::<EventV1>` or `bcs::from_bytes::<EventV2>`?

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+};
+use claims::{assert_matches, assert_ok, assert_some};
+use move_core_types::{
+    account_address::AccountAddress, ident_str, language_storage::ModuleId,
+    parser::parse_struct_tag, vm_status::AbortLocation,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize)]
+struct Aggregator {
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:22]: Why are these definitions required here? We didn't have to define these for other aggregator e2e tests so far right?

**Review by vusirikala — APPROVED** [review]

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/events_with_aggregators.move (hunk 1 above)
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/events_with_aggregators.move:15]: EventV2 is #[event] 
EventV1 is any object, but has an onchain handle. 

you can check out EventV2 AIP for more details

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs (hunk 2 above)
Comment (georgemitenkov) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:49]: Any should work, no specific reason why I used 0x1

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs (hunk 3 above)
Comment (georgemitenkov) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:42]: Yes, any account works

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs (hunk 5 above)
Comment (georgemitenkov) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:22]: So to test things and read a resource there are a few options:
1. What we did with aggregators tests, that is have a `check` transaction which does asserts.
2. What I did here: read a resource from state view and deserialize it into a rust struct (essentially there is a 1:1 mapping between Move values and this). Annotations allow to use bcs for serialization/deserialization.
3. Use view functions (haven't tried it before myself), but you can in theory define a view function which can give you certain values of a resource.

File: aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs (hunk 4 above)
Comment (georgemitenkov) [reply aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs:98]: Both EventV1 and EventV2 have the same structure, so I deserialize them into the same Rust struct

**Pull request merged**